iocore/net/quic/QUICFrameRecord.h

```diff
@@ -0,0 +1,43 @@
+/** @file
+ *
+ *  A brief file description
+ *
+ *  @section license License
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#include "QUICFrame.h"
+
+constexpr QUICFrameType QUIC_RECORD_FRAME[] = {
+  QUICFrameType::ACK,
+};
+
+struct QUICFrameInformation {
+  QUICFrameType type;
+  QUICEncryptionLevel level;
+
+  union {
+    // ack information 
+    struct QUICAckFrameInfo {
+      QUICPacketNumber largest_acknowledged;
+    } ack_frame_info;
+
```

Comment:
When a frame need to be tracked, you need to add an entry union and edit QUIC_RECORD_FRAME. I think it is might be ugly but not sure it is any better way here.

Reply:
How about making `frame_info` 8 byte opaque data? I'm not sure how many size is sufficient but QUICFrameGenerator (the parent class) doesn't need to know how each generators (sub classes) use the space. Some generators may use it as a 64 bit integer, and some other may use it as two 32 bit integers. They can use the field like that by casting it to their own internal structure.

Reply:
Not sure here since stream frame need fin flag and offset in the information. But we can round the offset up to reserve a bit to hold fin flag. That is pretty complex to user to use this api.

Reply:
Stream ID is also needed by stream's info.

Reply:
This is what I'm trying to say:

For example:
```
struct QUICFrameInformation {
.....
  uint8_t frame_info[16];  // Just an example. We may need more.
};
```

```
SomeGenerator : public QUICFrameGenerator
{
  struct MyFrameInfo {
    uint8_t fin_flag;
    uint32_t offset;
    uint32_t stream_id;
  }
.....
}

SomeGenerator::_on_frame_lost(QUICFrameInformation info)
{
  MyFrameInfo *info = reinterpret_cast<MyFrameInfo>(info.frame_info)
  if (info.fin_flag .....
}
```